### PR TITLE
Added new EventType (EInit)

### PR DESCRIPTION
**Pull request by sebHollersbacher**

### CODE REVIEW

File: plugins/org.eclipse.fordiac.ide.fbtypeeditor/src/org/eclipse/fordiac/ide/fbtypeeditor/editparts/TypeEditPart.java [hunk 1]
```diff
@@ -250,8 +256,8 @@ public INamedElement getINamedElement() {
 	@Override
 	public boolean isDirectEditable() {
 		// allow direct edit only for VarDeclarations and AdapterDeclarations
```
Comment (azoitl) [inline plugins/org.eclipse.fordiac.ide.fbtypeeditor/src/org/eclipse/fordiac/ide/fbtypeeditor/editparts/TypeEditPart.java]: please update the comment

File: plugins/org.eclipse.fordiac.ide.fbtypeeditor/src/org/eclipse/fordiac/ide/fbtypeeditor/editparts/TypeEditPart.java [hunk 2]
```diff
@@ -250,8 +256,8 @@ public INamedElement getINamedElement() {
...
-		return super.isDirectEditable()
-				&& (getCastedModel() instanceof VarDeclaration || getCastedModel() instanceof AdapterDeclaration);
+		return super.isDirectEditable() && (getCastedModel() instanceof VarDeclaration
```
Comment (azoitl) [inline plugins/org.eclipse.fordiac.ide.fbtypeeditor/src/org/eclipse/fordiac/ide/fbtypeeditor/editparts/TypeEditPart.java]: I think now the `&& ( ... )` can be removed as a whole as it covers all interface elements.

**Review by azoitl — CHANGES_REQUESTED** [review]

**Comment by oberlehner** [comment]
image
when an event is connected it cannot be changed.
This was an earlier feature, to prevent broken applications. But now the validation is mature enough to deal with these kind of errors, therefore please remove this check so that the event type can be changed here

**Comment by sebHollersbacher** [comment]
@oberlehner Has been changed to not consider Connections

**Comment by azoitl** [comment]
> @oberlehner Has been changed to not consider Connections

for me this is at least for the pin property sheet still broken.

**Review by oberlehner — APPROVED** [review]
Tested in UI and works fine !

**Review by azoitl — APPROVED** [review]

**Pull request merged**